code/datums/storage/storage.dm

```diff
@@ -0,0 +1,1029 @@
+/*!
+ * Datumized Storage
+ * Eliminates the need for custom signals specifically for the storage component, and attaches a storage variable (atom_storage) to every atom.
+ */
+
+///Helper proc to give something storage
+/atom/proc/create_storage(storage_type = /datum/storage, list/canhold, list/canthold, list/storage_type_limits)
+	RETURN_TYPE(/datum/storage)
+
+	if(atom_storage)
+		QDEL_NULL(atom_storage)
+
+	atom_storage = new storage_type(src)
```

Comment:
In jaeger code i think there was a manual qdel before calling create storage, so this would runtime because it gets delled twice